### PR TITLE
Update 3D tests

### PR DESCRIPTION
**Pull request by tkarna**
Updates for 3D model test cases

- adds support for quad meshes
- change `options.use_lax_friedrichs_tracer` to `False` by default
- updates for lock exchange and baroclinic eddies test

### CODE REVIEW

File: thetis/options.py [hunk 1]
```diff
@@ -372,9 +372,9 @@ class CommonModelOptions(FrozenConfigurable):
     use_lax_friedrichs_velocity = Bool(
         True, help="use Lax Friedrichs stabilisation in horizontal momentum advection.").tag(config=True)
     lax_friedrichs_velocity_scaling_factor = FiredrakeConstantTraitlet(
-        Constant(1.0), help="Scaling factor for Lax Friedrichs stabilisation term in horiozonal momentum advection.").tag(config=True)
+        Constant(1.0), help="Scaling factor for Lax Friedrichs stabilisation term in horizontal momentum advection.").tag(config=True)
     use_lax_friedrichs_tracer = Bool(
-        True, help="Use Lax Friedrichs stabilisation in tracer advection.").tag(config=True)
+        False, help="Use Lax Friedrichs stabilisation in tracer advection.").tag(config=True)
```
Comment (tkarna) [inline thetis/options.py]: My concern here is that this changes the default behavior for the 2D model as well. For 3D model, omitting the LF flux for tracers seems like the right thing to do, I'm not sure about 2D. If needed, we could specify this option separately for 2D and 3D. Any preferences?

File: examples/baroclinic_eddies/submitRuns.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
+"""
+Launches simulations on HPC machines.
+
+Run paramenters are inherited from the master test case. However, here all
```
Comment (stephankramer) [inline examples/baroclinic_eddies/submitRuns.py]: typo

File: thetis/options.py (hunk 1 above)
Comment (stephankramer) [reply thetis/options.py]: I think for the (limited number of) 2d cases with tracers that we've run so far, we've had LF switched off - as it's fairly diffusive. So I'm happy for this to be the default

File: examples/baroclinic_eddies/submitRuns.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+will excecute runs
+python mytestcase.py -r coarse -dt 25
+python mytestcase.py -r coarse -dt 50
+python mytestcase.py -r medium -dt 25
+python mytestcase.py -r medium -dt 50
+
+In addition to the options defined in the test case itself this script defines
+options:
+--test    - print job submission scirpts without launching them
```
Comment (stephankramer) [inline examples/baroclinic_eddies/submitRuns.py]: typo

File: examples/baroclinic_eddies/submitRuns.py [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+--dev     - run simulation in development queue for a short time for debugging
+--verbose - print job submission scirpts and debug information
+
+Depends on batchScriptLib
```
Comment (stephankramer) [inline examples/baroclinic_eddies/submitRuns.py]: What is that? To do with hpclauncher?

Comment (tkarna) [reply examples/baroclinic_eddies/submitRuns.py]: yes, it should say hpclauncher instead.

**Pull request merged**